### ISSUE TITLE
NoMethodError: undefined method `text' for nil:NilClass (forem/posts_controller.rb:17)

### ISSUE THREAD

**Issue by avitus** [rubysherpas/forem#515]
I'm puzzled as to what could cause this error. This is the relevant code:

```
def new
  @post = @topic.posts.build
  find_reply_to_post
  @post.text = view_context.forem_quote(@reply_to_post.text) if params[:quote]
end
```

Maybe topic was deleted before post was saved?

Trace is below:

```
/bundler/gems/forem-ef6a4679928c/app/controllers/forem/posts_controller.rb:17 in "new"
/gems/actionpack-4.0.2/lib/action_controller/metal/implicit_render.rb:4 in "send_action"
/gems/actionpack-4.0.2/lib/abstract_controller/base.rb:189 in "process_action"
/gems/actionpack-4.0.2/lib/action_controller/metal/rendering.rb:10 in "process_action"
/gems/actionpack-4.0.2/lib/abstract_controller/callbacks.rb:18 in "block in process_action"
/gems/activesupport-4.0.2/lib/active_support/callbacks.rb:513 in "_run__1203480121088887177__process_action__callbacks"
/gems/activesupport-4.0.2/lib/active_support/callbacks.rb:80 in "run_callbacks"
/gems/actionpack-4.0.2/lib/abstract_controller/callbacks.rb:17 in "process_action"
/gems/actionpack-4.0.2/lib/action_controller/metal/rescue.rb:29 in "process_action"
/gems/actionpack-4.0.2/lib/action_controller/metal/instrumentation.rb:31 in "block in process_action"
/gems/activesupport-4.0.2/lib/active_support/notifications.rb:159 in "block in instrument"
/gems/activesupport-4.0.2/lib/active_support/notifications/instrumenter.rb:20 in "instrument"
/gems/activesupport-4.0.2/lib/active_support/notifications.rb:159 in "instrument"
/gems/actionpack-4.0.2/lib/action_controller/metal/instrumentation.rb:30 in "process_action"
/gems/actionpack-4.0.2/lib/action_controller/metal/params_wrapper.rb:245 in "process_action"
/gems/activerecord-4.0.2/lib/active_record/railties/controller_runtime.rb:18 in "process_action"
/gems/actionpack-4.0.2/lib/abstract_controller/base.rb:136 in "process"
/gems/actionpack-4.0.2/lib/abstract_controller/rendering.rb:44 in "process"
/gems/actionpack-4.0.2/lib/action_controller/metal.rb:195 in "dispatch"
/gems/actionpack-4.0.2/lib/action_controller/metal/rack_delegation.rb:13 in "dispatch"
/gems/actionpack-4.0.2/lib/action_controller/metal.rb:231 in "block in action"
/gems/actionpack-4.0.2/lib/action_dispatch/routing/route_set.rb:80 in "call"
/gems/actionpack-4.0.2/lib/action_dispatch/routing/route_set.rb:80 in "dispatch"
/gems/actionpack-4.0.2/lib/action_dispatch/routing/route_set.rb:48 in "call"
/gems/actionpack-4.0.2/lib/action_dispatch/journey/router.rb:71 in "block in call"
/gems/actionpack-4.0.2/lib/action_dispatch/journey/router.rb:59 in "each"
/gems/actionpack-4.0.2/lib/action_dispatch/journey/router.rb:59 in "call"
/gems/actionpack-4.0.2/lib/action_dispatch/routing/route_set.rb:680 in "call"
/gems/railties-4.0.2/lib/rails/engine.rb:511 in "call"
/gems/railties-4.0.2/lib/rails/railtie/configurable.rb:30 in "method_missing"
/gems/actionpack-4.0.2/lib/action_dispatch/journey/router.rb:71 in "block in call"
/gems/actionpack-4.0.2/lib/action_dispatch/journey/router.rb:59 in "each"
/gems/actionpack-4.0.2/lib/action_dispatch/journey/router.rb:59 in "call"
/gems/actionpack-4.0.2/lib/action_dispatch/routing/route_set.rb:680 in "call"
/gems/rack-pjax-0.7.0/lib/rack/pjax.rb:12 in "call"
/gems/omniauth-1.1.4/lib/omniauth/strategy.rb:184 in "call!"
/gems/omniauth-1.1.4/lib/omniauth/strategy.rb:164 in "call"
/gems/newrelic_rpm-3.7.0.177/lib/new_relic/rack/error_collector.rb:50 in "call"
/gems/newrelic_rpm-3.7.0.177/lib/new_relic/rack/agent_hooks.rb:28 in "call"
/gems/newrelic_rpm-3.7.0.177/lib/new_relic/rack/browser_monitoring.rb:23 in "call"
/gems/warden-1.2.3/lib/warden/manager.rb:35 in "block in call"
/gems/warden-1.2.3/lib/warden/manager.rb:34 in "catch"
/gems/warden-1.2.3/lib/warden/manager.rb:34 in "call"
/gems/rack-1.5.2/lib/rack/etag.rb:23 in "call"
/gems/rack-1.5.2/lib/rack/conditionalget.rb:25 in "call"
/gems/rack-1.5.2/lib/rack/head.rb:11 in "call"
/gems/remotipart-1.2.1/lib/remotipart/middleware.rb:27 in "call"
/gems/actionpack-4.0.2/lib/action_dispatch/middleware/params_parser.rb:27 in "call"

...

```

**Comment by radar** [comment]
@avitus `@reply_to_post` is `nil`? That would only happen if `reply_to_id` was passed in and that ID wasn't a post for that topic... still, it shouldn't 500 in that case.

Can you please:
1. Show us some steps to reproduce this issue and,
2. Submit a patch to fix it?

Thanks!

**Comment by avitus** [comment]
I haven't yet been able to recreate it in my development environment. It happens fairly rarely in my production environment (about once a week) over 1000's of posts.

**Comment by alexcwatt** [comment]
I believe that the users are clicking to quote the post, but by the time they click it has already been deleted. #518 should fix.

**Comment by radar** [comment]
Fixed #518.
